components/frontend/src/metric/TrendGraph.js

```diff
@@ -1,20 +1,55 @@
+import { oneOf } from "prop-types"
 import { useContext } from "react"
+import { Placeholder, PlaceholderImage } from "semantic-ui-react"
 import { VictoryAxis, VictoryChart, VictoryLabel, VictoryLine, VictoryTheme } from "victory"
 
 import { DarkMode } from "../context/DarkMode"
 import { DataModel } from "../context/DataModel"
 import { measurementsPropType, metricPropType } from "../sharedPropTypes"
 import { capitalize, formatMetricScaleAndUnit, getMetricName, getMetricScale, niceNumber, scaledNumber } from "../utils"
+import { WarningMessage } from "../widgets/WarningMessage"
 
 function measurementAttributeAsNumber(metric, measurement, field, dataModel) {
     const scale = getMetricScale(metric, dataModel)
     const value = measurement[scale]?.[field] ?? null
     return value !== null ? Number(value) : null
 }
 
-export function TrendGraph({ metric, measurements }) {
+export function TrendGraph({ metric, measurements, loading }) {
     const dataModel = useContext(DataModel)
     const darkMode = useContext(DarkMode)
+    const chartHeight = 250
+    if (getMetricScale(metric, dataModel) === "version_number") {
+        return (
+            <WarningMessage
+                content="Trend graphs are not supported for metrics with a version number scale."
+                header="Trend graph not supported for version numbers"
+            />
+        )
+    }
+    if (loading === "failed") {
+        return (
+            <WarningMessage
+                content="Loading the measurements from the API-server failed."
+                header="Loading measurements failed"
+            />
+        )
+    }
+    if (loading === "loading") {
+        return (
+            <Placeholder fluid inverted={darkMode} style={{ height: chartHeight + 200 }}>
```

Comment:
Hmm, double magic number!

Reply:
Yeah, the chart height does not include the height of the title and other decorations so I had to estimate that. I'll make it a constant so it's clearer why the placeholder needs to be higher than the chart height.